### PR TITLE
Add `mygrad.any` and `Tensor.any`

### PR DESCRIPTION
**Pull request by samaocarpenter**
Addresses #218

### CODE REVIEW

File: src/mygrad/math/nondifferentiable.py [hunk 1]
```diff
@@ -45,3 +45,55 @@ def argmin(a, axis=None, out=None):
...
+        in which case a reference to `out` is returned.
+
+    See Also
+    --------
+    Tensor.any : equivalent method
+
+    """
+
+    a = a.data if isinstance(a, Tensor) else a
+    return np.any(a, axis, out, keepdims)
```
Comment (rsokl) [inline src/mygrad/math/nondifferentiable.py:99]: I recommend specifying the trailing arguments as keyword-arguments here.

File: tests/math/unary/test_nondifferentiable.py [hunk 2]
```diff
@@ -51,3 +51,23 @@ def test_argmax(a, data):
...
+
+
+@given(
+    a=hnp.arrays(
+        shape=hnp.array_shapes(max_side=4, max_dims=5), dtype=dtype_strat_numpy
+    ),
+    data=st.data(),
+)
+def test_any(a, data):
+    axis = data.draw(valid_axes(ndim=a.ndim, single_axis_only=True), label="axis")
```
Comment (rsokl) [inline tests/math/unary/test_nondifferentiable.py:63]: Why do you have `single_axis_only=True` here? We want to also test multiple axes

Comment (samaocarpenter) [reply tests/math/unary/test_nondifferentiable.py:63]: Honestly, I have almost no idea how to use Hypothesis...I just consulted a couple tutorials and mostly copy-pasted the tests from above. I'm working on learning it so I can write better tests in the future - thank you for pointing this out!

**Comment by samaocarpenter** [comment]
@rsokl I made those changes, thank you!

File: tests/math/unary/test_nondifferentiable.py [hunk 3]
```diff
@@ -51,3 +51,23 @@ def test_argmax(a, data):
...
+    tensor = Tensor(a)
+
+    # tensor input
+    assert_allclose(mg.any(tensor, axis=axis), np.any(a, axis=axis))
+
+    # tensor method
+    assert_allclose(tensor.any(axis=axis), a.any(axis=axis))
+
+    # array input
+    assert_allclose(mg.any(a, axis=axis), np.any(a, axis=axis))
```
Comment (rsokl) [inline tests/math/unary/test_nondifferentiable.py:73]: `keepdims` also needs to be exercised in these tests - this parameter should also be drawn by Hypothesis.

Comment (samaocarpenter) [reply tests/math/unary/test_nondifferentiable.py]: This should be resolved - thanks!

**Comment by rsokl** [comment]
Nice work @samaocarpenter ! Thanks for the contribution!

**Pull request merged**